### ISSUE TITLE
Wget.lua - Add dlls to compilator

### ISSUE THREAD

**Issue by tt-thoma** [zenith391/OCEmu#9]
I'm trying to download a file from github using wget (_`wget -f https://raw.githubusercontent.com/IgorTimofeev/MineOS/master/Installer/BIOS.lua /tmp/bios.lua`_) but wget say `Connection refused`
I tried to allow the emulator on my firewall, but still the same error
What can I do ?

_Edit_: pastebin works normally

**Comment by tt-thoma** [comment]
FINALLY !
I found why github is refusing connection
His DDos protection is blocking any requests send by the program
Now how to fix this ?

**Comment by zenith391** [comment]
Well to fix this I would need to know what's the precise error in the first place, if you can send that

**Comment by tt-thoma** [comment]
GitHub refuse any connection from the function `internet.request()`

**Comment by zenith391** [comment]
On my computer, `wget -f https://raw.githubusercontent.com/IgorTimofeev/MineOS/master/Installer/BIOS.lua /tmp/bios.lua` works correctly alongside using `internet.request()` for GitHub. Can you open https://raw.githubusercontent.com/IgorTimofeev/MineOS/master/Installer/BIOS.lua in your browser or does it only not work using `internet.request()` in OCEmu ?

**Comment by tt-thoma** [comment]
Have you tried to do the command in the emulator ?

**Comment by zenith391** [comment]
Yes, by on my computer I meant in OCEmu using OpenOS.

**Comment by tt-thoma** [comment]
Can I have your version of OCEmu ?
Your README.md is saying that you haven't compiled the current version yet

**Comment by misiuji** [comment]
I have the same issue on OCEmu. I think it's a windows issue, because zenith uses linux I think

![image](https://user-images.githubusercontent.com/46192406/123922102-fdfd9980-d987-11eb-8ca9-ab14579ce410.png)

also removing the `-f` option

![image](https://user-images.githubusercontent.com/46192406/123922496-6187c700-d988-11eb-9e01-ded095b75101.png)

**Comment by tt-thoma** [comment]
It's because @zenith391 have the last version

**Comment by tt-thoma** [comment]
After compiling it says "unsupported protocol"

**Comment by misiuji** [comment]
change `https://` to `http://`

**Comment by tt-thoma** [comment]
Maybe github refuse connection just because it wants https but there is no https protocol in \src\extras\socket

**Comment by misiuji** [comment]
maybe, here is the wget command in a normal terminal
![image](https://user-images.githubusercontent.com/46192406/123958705-aa05ab80-d9ad-11eb-8c97-e635eae173ca.png)

first request gets 301 Moved Permanently pointing to https url
secound request gets 200 ok

**Comment by tt-thoma** [comment]
@zenith391 Can you add an https addon ?

**Comment by tt-thoma** [comment]
![image](https://user-images.githubusercontent.com/72872981/123960465-60728c00-d9c0-11eb-8310-3adb1bf93608.png)
Opencomputers ^
OCemu V
![image](https://user-images.githubusercontent.com/72872981/123960761-be9f6f00-d9c0-11eb-982c-9ea8d21fe636.png)
The real Openos support https protocol.

**Comment by misiuji** [comment]
seems like the install script doesn't install luasec correctly
this would explain why this happens only on windows

**Comment by tt-thoma** [comment]
![image](https://user-images.githubusercontent.com/72872981/123962319-61a4b880-d9c2-11eb-808f-cfa11595bcf4.png)

**Comment by misiuji** [comment]
well if you look in /src/extras/ssl there is https.lua
that's where it shoud be
![image](https://user-images.githubusercontent.com/46192406/123962375-af64f500-d9b1-11eb-9d54-c3bb6c7155c1.png)

**Comment by tt-thoma** [comment]
![image](https://user-images.githubusercontent.com/72872981/123962767-da0b7980-d9c2-11eb-9695-1aa6dbcd3ee3.png)
I moved it and i'll try to recompilate

**Comment by misiuji** [comment]
![image](https://user-images.githubusercontent.com/46192406/123962697-07036080-d9b2-11eb-9e08-12971220b7c0.png)
this is how internet.lua gets https, but it doesn't find it

**Comment by tt-thoma** [comment]
Here we go again...

**Comment by misiuji** [comment]
> I moved it and i'll try to recompilate

recompiling will on do anything since the OCEmu.exe just calls boot.lua

**Comment by misiuji** [comment]
if you launch OCEmu via mingw then it works
![image](https://user-images.githubusercontent.com/46192406/123964656-f48a2680-d9b3-11eb-81ae-48414c1bfe6a.png)
![image](https://user-images.githubusercontent.com/46192406/123964668-f81dad80-d9b3-11eb-9297-8c323d352f9c.png)

**Comment by tt-thoma** [comment]
Ok

**Comment by misiuji** [comment]
I think it's not stripping the ssl core correctly, but I can be entirely wrong

**Comment by misiuji** [comment]
after requiring ssl at boot:

![image](https://user-images.githubusercontent.com/46192406/123972220-f3a8c300-d9ba-11eb-850a-65bee5613ee2.png)
at the bottom: "cannot find specified module"

manual boot from console works

**Comment by tt-thoma** [comment]
ffi file is missing
What is it ?

**Comment by misiuji** [comment]
it's a lua library https://luajit.org/ext_ffi.html

**Comment by tt-thoma** [comment]
Pls have you a tutorial to install I am stuck I have windows 64 bits I try following the readme nothing work I'm tired of solving problems

**Comment by misiuji** [comment]
for windows all you have to do is (once you installed msys2) run the script `msys2_setup_ocemu.sh` with `mingw64.exe` and hit enter a coupple of times, it should make a file `OCEmu.exe` in src dir

**Comment by misiuji** [comment]
there are no `libeay32.dll` and `ssleay32.dll` in any of the the bin folders
i think that the issue as per [this stackoverflow thread](https://stackoverflow.com/questions/40943901/using-luasec-produces-module-ssl-core-not-found)

**Comment by tt-thoma** [comment]
Thank you

**Comment by tt-thoma** [comment]
YEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEES

Thank you so munch

**Comment by tt-thoma** [comment]
How to do:
1. Download and install [MSYS2](https://www.msys2.org/)
2. Follow the steps
3. Go to https://www.dll-files.com/ and search for `libeay32.dll` and `ssleay32.dll`
4.  Download both
5. Extract both
6. Place both in `C:\msys64\usr\bin`. Use the custom path if you set one.
7. Type `MinGW` in the windows search bar, then launch it
8. In any folder you want, run: `git clone https://github.com/zenith391/OCEmu`
9. Go to the newly created OCEmu folder.
10. Run `./msys2_setup_ocemu.sh`. Sometimes you'll need to press enter

After that (considering that there is no error(s) while compiling), go to src, and you'll find the file OCEmu.exe

**Comment by misiuji** [comment]
now how to put this into the setup script @zenith391?
just download it from dll-files as a work around or some other way?

**Comment by fixator10** [comment]
So, after some digging in it with process hacker:
![image](https://user-images.githubusercontent.com/11073934/132848130-44556224-f235-4893-a5f0-0879b4a79cea.png)
`ssl.dll` by itself, most likely (based on its size), does nothing other than just loading and, probably, mapping SSL libs. The libraries itself are looked in PATH, and since MingW sets path to its own bin folder - its works in MingW. Doing `set PATH=%PATH%;C:\msys64\mingw64\bin` (assuming msys is installed to `C:\msys64`) before starting OCEmu fixes issue with HTTPS.

So basically, for standalone SSL support in OCEmu you need to: add folder with required libs to PATH or copy SSL libraries to PATH (or in folder with `ocemu.exe`, which is, basically, first place, where OS looking for files/libraries before looking for them in PATH)

Example of SSL after copying `libcrypto-1_1-x64.dll` and `libssl-1_1-x64.dll` to folder with OCEmu.exe:
![image](https://user-images.githubusercontent.com/11073934/132849557-1b7c9252-f1b2-495c-a689-677f35c8945d.png)
